site/app/templates/admin/Docker.twig

```diff
@@ -5,79 +5,144 @@
     {% if docker_info is not null %}
         {# TODO: include worker information here in the future #}
         <p>Docker Version : {{ docker_info['ServerVersion'] }} </p>
+        <p>Updated on: {{last_updated}}</p>
+        {% if error_logs | length > 0 %}
+            <textarea disabled="disabled" style="resize: none; width: 100%; height:25vh;">
+                {% for error_log in error_logs %}
+                    {{error_log ~ "\n"}}
+                {% endfor %}
+            </textarea>
+        {% endif %}
+        <button class="btn btn-primary" id="update-machines" onclick=updateImage("{{admin_url}}/update_docker") style="margin-top: 1em"> Update dockers and machines</button>
     {% endif %}
 </div>
 
 <div class="content">
-    <h2 class="docker-header">Autograding Docker Images</h2>
-    <button class="btn btn-default" 
-            onclick="collapseSection('autograding-images','autograding-images-collapse')"
-            id="autograding-images-collapse">
-            Collapse
-    </button>
-
-    <div id="autograding-images" style="display:block;">
-        <p class="black-message">These are the docker images that are installed and listed under autograding_containers.json</p>
-
-        <div class="docker-images-container">
-            {{ _self.renderDockerImage(autograding_containers['found']) }}
-        </div>
-    </div>
+    <h2 class="docker-header">Worker Machines</h2>
+    <table>
+        <tr>
+            <th></th>
```

Comment:
What is the reason for this empty header?

Reply:
This is an empty cell in the table, see screenshot. Just a personal preference in leaving it blank since that column seems intuitive, but I can also make it say Machine instead.
![image](https://user-images.githubusercontent.com/49821332/126578987-928de470-996e-4248-9fca-6daa0db441cd.png)

Reply:
I would suggest adding a table header for each column if it contains data, it may cause issues for screen readers otherwise or be less intuitive